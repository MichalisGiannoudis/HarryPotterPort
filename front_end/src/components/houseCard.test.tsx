import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import HouseCard from './houseCard';
import { House } from '@/models/house.interface';

const MOCK_CONTENT = {
  pageId: 'mock-page',
  founderLabel: 'founder-label',
  houseSearchLabel: 'house-search-label',
  houseSearchEmptyResultsLabel: 'house-empty-result-label',
  houseSearchErrorLabel: 'error-label',
  houseTraitSearchLabel: 'house-trait-search-label',
}

const MOCK_HOUSE_CARD_PROPS = {
  founderLabel: 'founder-label',
  houseTraitSearchLabel: 'house-trait-search-label',
}

const mockHouse: House = {
  id: '1',
  name: 'Gryffindor',
  animal: 'Lion',
  houseColours: 'Scarlet and Gold',
  founder: 'Godric Gryffindor',
  element: 'Fire',
  ghost: 'Nearly Headless Nick',
  commonRoom: 'Gryffindor Tower',
  heads: [
    {
      id: '1',
      firstName: 'Minerva',
      lastName: 'McGonagall'
    }
  ],
  traits: [
    { id: '1', name: 'Bravery' },
    { id: '2', name: 'Courage' },
    { id: '3', name: 'Chivalry' },
  ],
};

jest.mock('@/hooks/useDebounce', () => ({
  useDebounce: (value: string) => value,
}));

const testRender = (...props: any[]) => render(<HouseCard house={mockHouse}  {...MOCK_HOUSE_CARD_PROPS} {...props} />)

describe('HouseCard', () => {
  it('renders house information correctly', () => {
    const { container, getByTestId } = testRender();

    expect(screen.getByText('Gryffindor')).toBeInTheDocument();
    expect(screen.getByText('Lion')).toBeInTheDocument();
    expect(screen.getByText('founder-label:')).toBeInTheDocument();
    expect(screen.getByText('Godric Gryffindor')).toBeInTheDocument();
    expect(getByTestId('house-card-' + mockHouse.id)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('filters traits based on search input', () => {
    const { getByTestId } = testRender();

    const searchInput = getByTestId(`house-card-${mockHouse.id}-trait-search`);
    fireEvent.change(searchInput, { target: { value: 'bra' } });

    expect(screen.getByText('Bravery')).toBeInTheDocument();
    expect(screen.queryByText('Courage')).not.toBeInTheDocument();
    expect(screen.queryByText('Chivalry')).not.toBeInTheDocument();
  });
}); 