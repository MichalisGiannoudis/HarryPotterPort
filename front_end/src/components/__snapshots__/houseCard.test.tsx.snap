// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HouseCard renders house information correctly 1`] = `
<div>
  <div
    class="flex flex-col gap-2 max-w-sm rounded-lg border border-gray-300 shadow-xl bg-white p-4"
    data-testid="house-card-1"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <h2
        class="text-2xl font-bold font-f"
      >
        Gryffindor
      </h2>
      <span
        class="text-md font-semibold"
      >
        Lion
      </span>
    </div>
    <div>
      <div
        class="h-5 rounded-md"
        style="background: linear-gradient(to right, white, black);"
      />
    </div>
    <p
      class="mb-2"
    >
      <span
        class="font-semibold"
      >
        founder-label
        : 
      </span>
      <span
        class="font-bold"
      >
        Godric Gryffindor
      </span>
    </p>
    <input
      class="w-[65%] mb-3 px-3 py-1 border rounded-xl focus:outline-none focus:ring-gray-300 focus:border-gray-300 border-gray-200"
      data-testid="house-card-1-trait-search"
      placeholder="house-trait-search-label"
      type="text"
      value=""
    />
    <div
      class="flex flex-wrap gap-2"
    >
      <span
        class="bg-gray-800 text-white flex items-center justify-center px-3 rounded-lg text-sm"
      >
        Bravery
      </span>
      <span
        class="bg-gray-800 text-white flex items-center justify-center px-3 rounded-lg text-sm"
      >
        Courage
      </span>
      <span
        class="bg-gray-800 text-white flex items-center justify-center px-3 rounded-lg text-sm"
      >
        Chivalry
      </span>
    </div>
  </div>
</div>
`;
